new_webapp/src/actions/index.js

```diff
@@ -165,4 +166,21 @@ export const getMoreExamples = ({candidateId, code, usages}) => (dispatch, getSt
                 message: errorResult.error || "Unknown Error",
             }));
         })
-}
\ No newline at end of file
+}
+
+// This is where a request needs to be sent to the server
+// id: uuid
+export const doStop = ({id}) => (dispatch) => {
+    dispatch(setSearchStatus({status:DONE}));
+    Search.sendStopSignal({id})
+    .then(result => {
+        return dispatch(stopSearch({id}));
+		})
+    .catch(error => {
+        return dispatch(setSearchStatus({
+            status: ERROR,
+            errorMessage: error.message
+        }));
+    });
+    return;
```

Comment:
oh sorry, I should have caught this earlier: you should return the promise! It helps with chaining.